File: src/main/java/com/moira/itda/domain/signup/controller/SignupController.java
package com.moira.itda.domain.signup.controller;

import com.moira.itda.domain.signup.dto.request.SignupRequest;
import com.moira.itda.domain.signup.service.SignupService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class SignupController {
    private final SignupService signupService;

    /**
     * 회원가입 > 닉네임 중복 확인
     */
    @GetMapping("/api/signup/check/nickname")
    public ResponseEntity<Object> checkNickname(String nickname) {
        signupService.checkNickname(nickname);

        return ResponseEntity.ok(null);
    }

    /**
     * 회원가입 > 이메일 중복 확인
     */
    @GetMapping("/api/signup/check/email")
    public ResponseEntity<Object> checkEmail(String email) {
        signupService.checkEmail(email);

        return ResponseEntity.ok(null);
    }

    /**
     * 회원가입 > 회원가입
     */
    @PostMapping("/api/signup")
    public ResponseEntity<Object> signup(@RequestBody SignupRequest request) {
        signupService.signup(request);

        return ResponseEntity.ok(null);
    }
}

File: src/main/kotlin/com/moira/itda/domain/user/signup/controller/SignupController.kt
package com.moira.itda.domain.user.signup.controller

import com.moira.itda.domain.user.signup.dto.request.SignupRequest
import com.moira.itda.domain.user.signup.service.SignupService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

/**
 * 회원가입 페이지
 */
@RestController
class SignupController(
    private val signupService: SignupService
) {
    /**
     * 회원가입 > 닉네임 중복 확인
     */
    @GetMapping("/api/signup/check/nickname")
    fun checkNickname(@RequestParam nickname: String): ResponseEntity<Nothing> {
        signupService.checkNickname(nickname = nickname)

        return ResponseEntity.ok(null)
    }

    /**
     * TODO: 회원가입 > 이메일 본인인증
     */
    @GetMapping("/api/signup/identify")
    fun checkEmail(@RequestParam email: String): ResponseEntity<Nothing> {
        signupService.identify(email = email)

        return ResponseEntity.ok(null)
    }

    /**
     * 회원가입
     */
    @PostMapping("/api/signup")
    fun signup(@RequestBody request: SignupRequest): ResponseEntity<Nothing> {
        signupService.signup(request = request)

        return ResponseEntity.status(HttpStatus.CREATED).body(null)
    }
}